Mathlib/Data/Nat/Digits.lean

```diff
@@ -442,6 +442,18 @@ theorem le_digits_len_le (b n m : ℕ) (h : n ≤ m) : (digits b n).length ≤ (
   monotone_nat_of_le_succ (digits_len_le_digits_len_succ b) h
 #align nat.le_digits_len_le Nat.le_digits_len_le
 
+@[mono]
+theorem ofDigits_monotone {p q : ℕ} (L : List ℕ) (h : p ≤ q) : ofDigits p L ≤ ofDigits q L := by
+  induction' L with _ _ hi
+  · rfl
+  · simp only [ofDigits, cast_id, add_le_add_iff_left]
+    exact Nat.mul_le_mul h hi
+
+theorem digit_sum_le (p n : ℕ) (h : 1 ≤ p) : List.sum (digits p n) ≤ n := by
```

Comment:
One last annoying remark, I think this should still be (trivially) true when `p = 0` as then `digits p n = [n]`, so it would be best to make this a proof by cases of the general statement and avoid propogating the `1 \le p` hypothesis downstream

Reply:
That's great! I removed the hypothesis `(h : 1 ≤ p)`.